main/src/ca/uwaterloo/flix/language/phase/Redundancy.scala

```diff
@@ -706,6 +706,47 @@ object Redundancy {
     case Expression.Par(exp, _) =>
       visitExp(exp, env0, rc)
 
+    case Expression.ParYield(frags, exp, _, _, _, _) =>
+      val (used, env1, fvs) = frags.foldLeft((Used.empty, env0, Set.empty[Symbol.VarSym])) {
+        case ((usedAcc, envAcc, fvsAcc), ParYield.Fragment(p, e, _)) =>
```

Comment:
It might be nice to pull this into a helper. I feel like there's a lot going on here.